lexer/tiger.lex

```diff
@@ -1,18 +1,125 @@
-type pos = int
+type pos = int;
 type lexresult = Tokens.token
 
 val lineNum = ErrorMsg.lineNum
 val linePos = ErrorMsg.linePos
-fun err(p1,p2) = ErrorMsg.error p1
 
-fun eof() = let val pos = hd(!linePos) in Tokens.EOF(pos,pos) end
+val partialString = ref ""
+val stringStartPos = ref ~1
 
+val commentCount = ref 0
 
-%% 
+fun getEscapedChar c =
+	case c of
+		  "\\^@" => Char.toString(chr(0))
+		| "\\^G" => Char.toString(chr(7))
+		| "\\^H" => Char.toString(chr(8))
+		| "\\^I" => Char.toString(chr(9))
+		| "\\^J" => Char.toString(chr(10))
+		| "\\^K" => Char.toString(chr(11))
+		| "\\^L" => Char.toString(chr(12))
+		| "\\^M" => Char.toString(chr(13))
+		| "\\^Z" => Char.toString(chr(26))
+		| "\\^[" => Char.toString(chr(27))
+		| "\\n"  => Char.toString(chr(10))
+		| "\\t"  => Char.toString(chr(9))
+		| "\\\"" => Char.toString(chr(34))
+		| "\\"	 => Char.toString(chr(92))
+		| a => a
+
+fun eof() = let val pos = hd(!linePos)
+	    in
+			(if (!stringStartPos) >= 0 then ErrorMsg.error (!stringStartPos) (" open string at end of file") else ());
+			(if (!commentCount) > 0 then ErrorMsg.error pos (" open comment at end of file") else ());
+			Tokens.EOF(pos,pos)
+	    end
+%%
+%s COMMENT_STATE STRING_STATE;
+escapeDigits = ([0-1][0-9][0-9])|(2[0-4][0-9])|(25[0-5]);
```

Comment:
This should be Octal instead of Decimal?

Reply:
See above